Master values hack

manorman5 opened the pull request:
hey! I was having trouble extracting values csvs for both point estimates and variances. Specifically, I ran into three different errors. 
a) output_format not found -- change on line 118 of lib/results.py
b) years not found -- change on lines 147-151, 142 and 127 of lib/results.py and line 36 of quantiles.py
c) deltamethod_variance not found -- change on line 69 of quantiles.py
I felt like my solutions to a and c weren't particularly hacky, but my solution to b was very hacky. Let me know if you have questions.

Commit:
allow quantiles.py to retrieve years from sum_into_data, and allow results.sum_into_data to return years as well as data. this allowed me to circumvent the issue I was facing. I do not understand the implications of my work around, but my results match old results before sum_into_data started to get used. i also added in a small change to line 118 of results.py to deal with the fact that output_format is not defined in results.py.

manorman5 committed:
results. missing

missing package for function call

jrising commented on gcp/extract/lib/results.py:
```diff
@@ -124,7 +124,7 @@ def sum_into_data(root, basenames, columns, config, transforms, vectransforms):
                         observations += 1
                         continue
-                
+                    years_for_return = years                
```
This isn't used. Please remove.

manorman5 replied:
done. sorry this was leftover from an earlier hack I tried and clearly forgot about.

jrising commented on gcp/extract/lib/results.py:
```diff
@@ -139,12 +139,16 @@ def sum_into_data(root, basenames, columns, config, transforms, vectransforms):
+    
+    do_deltamethod = False if configs.is_parallel_deltamethod(config) else config.get('deltamethod', None)
+    filepath = os.path.join(last_targetdir, basenames[0] + '.nc4')
+    years, regions, data_not = bundles.read(filepath = filepath, column = columns[0], deltamethod = do_deltamethod)
```
I don't think we need this. Either the `try` clause above successfully ran, in which case `years` is already defined; or it didn't, in which case this would produce an error. I suggest putting `years = []` at the top of this function, removing these lines, and just using whatever the `years` variable ends up being set to.

jrising replied:
Also drop the `last_targetdir` line above, if you go this route.

manorman5 replied:
makes a lot more sense 💯

manorman5 replied:
so, the logic here is that in python variables still exist from the loop they were generated in after the loop ends. I can google this for more info. just learning out loud in case this is obviously wrong.

jrising commented on gcp/extract/quantiles.py:
```diff
@@ -33,7 +33,7 @@
 
 # Collect all available results
-data = results.sum_into_data(config['results-root'], basenames, columns, config, transforms, vectransforms)
+data, years = results.sum_into_data(config['results-root'], basenames, columns, config, transforms, vectransforms)
```
Also add `, parallel_deltamethod_years` to the other use of `sum_into_data` (line 41) or that will break.

manorman5 replied:
done.

jrising commented on gcp/extract/quantiles.py:
```diff
@@ -33,7 +33,7 @@
 
 # Collect all available results
-data = results.sum_into_data(config['results-root'], basenames, columns, config, transforms, vectransforms)
+data, years = results.sum_into_data(config['results-root'], basenames, columns, config, transforms, vectransforms)
```
Also add `, years` to the other use of `sum_into_data` (line 41) or that will break.

manorman5 replied:
i'm assuming the `, parallel_deltamethod_years` fix above is preferred over `, years` in line 41

jrising replied:
Yes, thanks!

jrising reviewed (changes requested):
Thanks for these fixes! It looks like they all came from when I moved the code that was in `sum_into_data` into its own function. Just a few further edits requested.

manorman5 commented:
hey @TomBearpark and @jrising, I'm out next week, so if there are any updates to this pull request thread I want to get tom looped in. I'll address your comments hopefully later today. Thanks!

manorman5 committed:
updating with James's requests

lesson learned: variables defined in loop remain in scope after the loop exits

manorman5 committed:
adding in comment 

for future troubleshooter -- define function return variables at top of function so if variables go undefined function doesnt boot (in case this is a desired feature)

manorman5 committed:
add parallel_deltamethod_years

didn't think through implications of my changes to sum_into_data throughout the entire quantiles.py code

manorman5 commented:
hi @jrising , okay I think i've resolved all of your requests. Yep, moving code from quantiles.py into sum_into_data seemed to have caused pretty much all the bugs. Thanks for the feedback! I unfortunately can't test the code, but I think the logic flows.

jrising committed:
merge in master

jrising reviewed (approved):
I merged the changes from master into this branch and resolved the commits. Can you test it to make sure that I didn't break anything in the process? If it's all good, we can merge it in.

manorman5 commented:
amazing thanks James! either I or @TomBearpark will test it out when the servers come back up.

manorman5 commented:
hey @jrising! I tested this it worked. Let me know if you have questions. Sorry if I wasn't supposed to merge it into master.

jrising commented:
Great. Thanks!
